tests/ui/NavigationTest.js

```diff
@@ -0,0 +1,216 @@
+        // await new Promise((resolve) => setTimeout(resolve, CONST.ANIMATED_TRANSITION));
+
+        // Verify there is now one option in the sidebar
+        optionRows = screen.queryAllByAccessibilityHint(optionRowsHintText);
+        expect(optionRows).toHaveLength(1);
+        screen.getByText(`Hi from ${ALICE.NAME}`);
+
+        await navigateToSidebarOption(0);
+        const welcomeMessageHintText = Localize.translateLocal('accessibilityHints.chatWelcomeMessage');
+        screen.getByLabelText(welcomeMessageHintText);
+        const reportCommentsHintText = Localize.translateLocal('accessibilityHints.chatMessage');
+        let reportComments = screen.getAllByLabelText(reportCommentsHintText);
+        expect(reportComments).toHaveLength(1);
+        screen.getByText(`Comment 1 from ${ALICE.NAME}`);
+
+        // Add another report to Onyx
+        now = DateUtils.getDBTime();
+        await Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${2}`, {
+            reportID: '2',
+            reportName: CONST.REPORT.DEFAULT_REPORT_NAME,
+            lastReadTime: now,
+            lastVisibleActionCreated: now,
+
+            // Note: this is intentionally different from the actual last message to prevent the sidebar from matching when we want to make sure that the central report pane is visible
+            lastMessageText: `Hi from ${BOB.NAME}`,
+            participantAccountIDs: [ALICE.ACCOUNT_ID, BOB.ACCOUNT_ID],
+            type: CONST.REPORT.TYPE.CHAT,
+        });
+        createdReportActionID = NumberUtils.rand64();
+        await Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${2}`, {
+            [createdReportActionID]: {
+                actionName: CONST.REPORT.ACTIONS.TYPE.CREATED,
+                automatic: false,
+                created: now,
+                reportActionID: createdReportActionID,
+                message: [
+                    {
+                        style: 'strong',
+                        text: '__FAKE__',
+                        type: 'TEXT',
+                    },
+                    {
+                        style: 'normal',
+                        text: 'created this report',
+                        type: 'TEXT',
+                    },
+                ],
+            },
+            2: TestHelper.buildTestReportComment(now, BOB.ACCOUNT_ID, '2', BOB.NAME),
+        });
+        await waitForBatchedUpdates();
+
+        // Message from alice should still be visible
+        screen.getByText(`Comment 1 from ${ALICE.NAME}`);
+
+        reportComments = screen.queryByText(`Comment 2 from ${BOB.NAME}`);
+        expect(reportComments).toBeNull();
+
+        // Replace report 1 with report 2 in the stack
+        Navigation.replaceReportIDInNavigationStack('1', '2');
+        await waitForBatchedUpdates();
+
+        // Report 2 should now be visible, and report 1 should not
+        screen.getByText(`Comment 2 from ${BOB.NAME}`);
```

Comment:
Should we do something like: 

```js
expect(screen.queryByText(`Comment 1 from ${ALICE.NAME}`)).toBeNull()
```

Reply:
Good catch, and interestingly that assertion is failing. I'm not sure _why_ but also it's pretty hard to debug